File: PostRequest.java
import java.io.DataOutputStream;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;

public class PostRequest {
    public static void main(String[] args) {
        try {
            // Define the URL for the POST request
            URL url = new URL("http://127.0.0.1:8000/chats");
            HttpURLConnection con = (HttpURLConnection) url.openConnection();

            // Set the request method to POST
            con.setRequestMethod("POST");
            con.setRequestProperty("Content-Type", "application/json; utf-8");
            con.setRequestProperty("Accept", "application/json");
            con.setDoOutput(true);

            // Create JSON request body
            String jsonInputString = "{\"sender\": \"aran\", \"content\": \"can I plant maize\"}";

            // Send the JSON input string
            try (DataOutputStream out = new DataOutputStream(con.getOutputStream())) {
                out.writeBytes(jsonInputString);
                out.flush();
            }

            // Get the response code
            int status = con.getResponseCode();

            // Read the response
            BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream(), StandardCharsets.UTF_8));
            String inputLine;
            StringBuilder content = new StringBuilder();
            while ((inputLine = in.readLine()) != null) {
                content.append(inputLine);
            }

            // Close the connections
            in.close();
            con.disconnect();

            // Print the response content
            System.out.println(content.toString());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: PostRequest.kt
import java.io.BufferedReader
import java.io.DataOutputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL
import java.nio.charset.StandardCharsets

fun main() {
    try {
        // Define the URL for the POST request
        val url = URL("http://127.0.0.1:8000/chats")
        val con = url.openConnection() as HttpURLConnection

        // Set the request method to POST
        con.requestMethod = "POST"
        con.setRequestProperty("Content-Type", "application/json; utf-8")
        con.setRequestProperty("Accept", "application/json")
        con.doOutput = true

        // Create JSON request body
        val jsonInputString = """{"sender": "aran", "content": "can I plant maize"}"""

        // Send the JSON input string
        DataOutputStream(con.outputStream).use { out ->
            out.writeBytes(jsonInputString)
            out.flush()
        }

        // Get the response code
        val status = con.responseCode

        // Read the response
        BufferedReader(InputStreamReader(con.inputStream, StandardCharsets.UTF_8)).use { reader ->
            val content = StringBuilder()
            var inputLine: String?
            while (reader.readLine().also { inputLine = it } != null) {
                content.append(inputLine)
            }

            // Print the response content
            println(content.toString())
        }

        // Close the connection
        con.disconnect()
    } catch (e: Exception) {
        e.printStackTrace()
    }
}
